### PR TITLE
Fix #21 - Support '+=' operator

### LINKED ISSUE #21: Support '+=' operator
``` sh
i := 0;
i += 5;
```

gives parser errors:

```
test.boa: compilation failed: Encountered parser error "[@4,8:8='i',<76>,3:0]" at line 3, column 0. error: ';' expected
i += 5;
^
    at unknown stack
test.boa: compilation failed: Encountered parser error "[@5,10:10='+',<55>,3:2]" at line 3, column 2. error: ';' expected
i += 5;
  ^
    at unknown stack
test.boa: compilation failed: Encountered parser error "[@6,11:11='=',<67>,3:3]" at line 3, column 3. no viable alternative at input '+='
i += 5;
   ^
```

### PR DESCRIPTION
**Pull request by CheShianHung**
Implement support for #21 - the +=, *=, /=, -= etc assignment variants.

### CODE REVIEW

File: src/java/boa/compiler/transforms/LocalAggregationTransformer.java [hunk 1]
```diff
@@ -126,6 +127,7 @@ protected void generateStoreValue(final EmitStatement n) {
 		n.replaceStatement(n,
 			new AssignmentStatement(
 				new Factor(id.clone()),
+				"=",
```
Comment (psybers) [inline src/java/boa/compiler/transforms/LocalAggregationTransformer.java]: This class does not need changed.  Instead, modify the AssignmentStatement class to have a 2 argument constructor that defaults to the '=' operator.

File: src/java/boa/compiler/visitors/CodeGeneratingVisitor.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright 2017, Anthony Urso, Hridesh Rajan, Robert Dyer, Ramanathan Ramu,
+ *				   Che Shian Hung
```
Comment (psybers) [inline src/java/boa/compiler/visitors/CodeGeneratingVisitor.java]: move name up to end of previous line

File: src/antlr/Boa.g [hunk 3]
```diff
@@ -421,7 +421,7 @@ fixpStatement returns [FixPStatement ast]
 	@after { $ast.setPositions($l, $c, getEndLine(), getEndColumn()); }
 	: (LPAREN id1=identifier COMMA id2=identifier COLON t=identifier RPAREN { $ast = new FixPStatement(); $ast.setParam1((Component)new Component($id1.ast, $t.ast).setPositions($id1.ast.beginLine, $id1.ast.beginColumn, $t.ast.endLine, $t.ast.endColumn));$ast.setParam2((Component)new Component($id2.ast, $t.ast).setPositions($id2.ast.beginLine, $id2.ast.beginColumn, $t.ast.endLine, $t.ast.endColumn));} (COLON rt=type {$ast.setReturnType($rt.ast);}) (s=programStatement { $ast.setBody($s.ast); }))
 	;
-
```
Comment (psybers) [inline src/antlr/Boa.g]: revert any whitespace only changes

**Comment by CheShianHung** [comment]
All change requests are done. All the test cases compile successfully and output the right result, but it seems that I break other stuff.

**Comment by psybers** [comment]
None of our current test cases test 'result' (aka evaluate).

**Comment by psybers** [comment]
Regarding storing op as text, this mirrors how we handle ops in all the other node types.

**Comment by nguyenhoan** [comment]
I added all three as general comments.
Please feel free to take any or none as change request.

**Comment by nguyenhoan** [comment]
I guess emit is a statement while left shift is an expression.

Adding test cases would be good.

**Pull request merged**